beacon_node/eth1/src/service.rs

```diff
@@ -121,6 +122,20 @@ impl EndpointsCache {
         state
     }
 
+    /// Update the endpoint_index to index
+    /// Returns previous value
+    pub fn update_cur_endpoint_url(&self, url: SensitiveUrl) -> (Option<SensitiveUrl>, bool) {
+        let mut rw_ref = self.cur_endpoint_url.write();
+        let prev_url = rw_ref.clone();
+        *rw_ref = Some(url.clone());
+        let changed: bool = if let Some(purl) = &prev_url {
+            purl.full != url.full
+        } else {
+            true
```

Comment:
Here we're going to create an "endpoint changed" log each time we startup. Was that what you had intended? I thought that might be a little noisy but I'm open to it.

Over in https://github.com/paulhauner/lighthouse/commit/d917c5d7650280e02f47380fd084528d1a45a0ac I've made a suggestion for one that won't log the message on startup. Even if you want to log on startup, you might find the `map_or` function useful for the `if let Some(_) = _ { ... }  else { true }` pattern :)